CMakeLists.txt

```diff
@@ -353,91 +356,112 @@ if (OPENAL)
   endif()
 endif()
 
+set(BASE_DATA_DIR "./" CACHE INTERNAL "Base data dir")
 
-if (NOT APPLE)
-#install(TARGETS barony
-#  RUNTIME DESTINATION bin
-#  COMPONENT Runtime
-#)
+if (NOT APPLE AND UNIX)
+include(GNUInstallDirs)
+if (DATA_DIR)
+	set(BASE_DATA_DIR "${CMAKE_INSTALL_FULL_DATAROOTDIR}/barony")
+endif()
+message(STATUS "Base data directory ${BASE_DATA_DIR}")
+
+install(TARGETS barony
+	RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR}
+  COMPONENT Runtime
+)
 else()
 install(TARGETS barony
   RUNTIME DESTINATION bin COMPONENT Runtime BUNDLE DESTINATION Resources
 )
 endif()
 
+add_definitions(-DBASE_DATA_DIR=\"${BASE_DATA_DIR}\")
+
 if (APPLE)
   #add_executable(editor OSX/SDLmain.m ${EDITOR_SOURCES})
   #set(MACOSX_BUNDLE_ICON_FILE "${PROJECT_SOURCE_DIR}/editor.icns")
   #add_executable(editor MACOSX_BUNDLE OSX/SDLmain.m ${EDITOR_SOURCES} ${COPY_FRAMEWORKS} libfmodex.dylib  /opt/local/lib/libpng16.16.dylib)
   #add_executable(editor MACOSX_BUNDLE OSX/SDLmain.m ${EDITOR_SOURCES} libfmodex.dylib  /opt/local/lib/libpng16.16.dylib)
   if (FMOD_ENABLED)
-    add_executable(editor MACOSX_BUNDLE ${EDITOR_SOURCES} libfmodex.dylib  /opt/local/lib/libpng16.16.dylib)
+	  add_executable(${EDITOR_EXE_NAME} MACOSX_BUNDLE ${EDITOR_SOURCES} libfmodex.dylib  /opt/local/lib/libpng16.16.dylib)
   else()
-    add_executable(editor MACOSX_BUNDLE ${EDITOR_SOURCES} /opt/local/lib/libpng16.16.dylib)
+	  add_executable(${EDITOR_EXE_NAME} MACOSX_BUNDLE ${EDITOR_SOURCES} /opt/local/lib/libpng16.16.dylib)
   endif()
   #SET_SOURCE_FILES_PROPERTIES(${MACOSX_BUNDLE_ICON_FILE} PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
   set_source_files_properties(${EDITOR_SOURCES} PROPERTIES COMPILE_FLAGS "-x objective-c++")
 else()
-  add_executable(editor ${EDITOR_SOURCES})
+	add_executable(${EDITOR_EXE_NAME} ${EDITOR_SOURCES})
 endif()
 
 if(WIN32)
-  target_link_libraries(editor -lmingw32)
-  target_link_libraries(editor -lSDL2main)
-  target_link_libraries(editor -lSDL2)
-  target_link_libraries(editor -lSDL2_net)
-  target_link_libraries(editor -lSDL2_image)
-  target_link_libraries(editor -lSDL2_ttf)
+	target_link_libraries(${EDITOR_EXE_NAME} -lmingw32)
+	target_link_libraries(${EDITOR_EXE_NAME} -lSDL2main)
+	target_link_libraries(${EDITOR_EXE_NAME} -lSDL2)
+	target_link_libraries(${EDITOR_EXE_NAME} -lSDL2_net)
+	target_link_libraries(${EDITOR_EXE_NAME} -lSDL2_image)
+	target_link_libraries(${EDITOR_EXE_NAME} -lSDL2_ttf)
```

Comment:
This could all be one `target_link_libraries` call.

Reply:
true I just lazily kept that way ... will correct all in a jiffy :-)